internal/equality/equality_test.go

```diff
@@ -42,6 +43,40 @@ var (
 	}
 )
 
+// New makes a Gateway object using the provided ns/name for the object's
+// ns/name.
+func newGateway(ns, name string) *gatewayv1alpha1.Gateway {
```

Comment:
`newGateway` is nearly identical to `New()` in the internal/gateway pkg. I plan to move equality to a public pkg in the future, so I would like to avoid adding any internal deps.